### PR TITLE
Modernize and simplify the setup

### PR DESCRIPTION
**Pull request by sarlinpe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,20 +70,21 @@ T_w_i = sessions.trajectories[keys[0]]  # first pose, from sensor/rig to world
 
 - Python >= 3.8
 - [hloc](https://github.com/cvg/Hierarchical-Localization) and its dependencies, including [COLMAP](https://colmap.github.io/install.html) built from source
-- everything listed in `requirements/lamar.txt` installed with
+
+:two: Install the LaMAR libraries and pull the remaining pip dependencies:
 ```bash
-python -m pip install -r requirements/lamar.txt
+python -m pip install -e .
```
Comment (pablospe) [inline README.md:76]: Isn't it doing more than that (installing everything listed in `requirements/lamar.txt`)? Isn't this install it in development mode with the `-e`?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "Operating System :: OS Independent",
+]
+dependencies = [
+    "numpy",
+    "scipy",
+    "h5py",
+    "torch>=1.1",
+    "tqdm>=4.36.0",
+    "opencv-python",
+    #"pycolmap @ git+https://github.com/colmap/pycolmap.git@v0.4.0",
```
Comment (pablospe) [inline pyproject.toml:21]: What happen if these lines are uncommented? Just to avoid they were left commented accidentally.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+[project.optional-dependencies]
+scantools = [
+    "matplotlib",
+    "lxml",
+    "beautifulsoup4==4.10.0",
+    "open3d",
+    "pytijo",
+    "plyfile",
+    "astral",
```
Comment (pablospe) [inline pyproject.toml:34]: astral?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    #"pycolmap @ git+https://github.com/colmap/pycolmap.git@v0.4.0",
+    #"pyceres @ git+https://github.com/cvg/pyceres.git@v1.0",
+]
+urls = {Repository = "https://github.com/microsoft/lamar-benchmark"}
+
+[project.optional-dependencies]
+scantools = [
+    "matplotlib",
+    "lxml",
+    "beautifulsoup4==4.10.0",
```
Comment (pablospe) [inline pyproject.toml:30]: I remembered to saw a warning about beautifulsoup4 and xml. I will see if I can replicate what I am saying.

File: README.md (hunk 1 above)
Comment (sarlinpe) [reply README.md:76]: 1. Installing the package now automatically pulls the dependencies. This makes it possible to run `pip install git+https://github.com/microsoft/lamar-benchmark.git` when we don't want a local copy.
2. `-e` enables editable mode (needed if the users need to edit the code, most likely if extending it).

File: pyproject.toml (hunk 2 above)
Comment (sarlinpe) [reply pyproject.toml:21]: My bad, indeed shouldn't have been commented out. Now fixed.

File: pyproject.toml (hunk 3 above)
Comment (sarlinpe) [reply pyproject.toml:34]: https://github.com/microsoft/lamar-benchmark/blob/ee0d374a6f5f3910f7224e86b9bc0bb36ed8cd57/scantools/utils/tagging.py#L2-L3

File: pyproject.toml (hunk 4 above)
Comment (sarlinpe) [reply pyproject.toml:30]: This version was pinned in the `requirements.txt`, haven't encountered any problem.

**Review by pablospe — APPROVED** [review]

**Pull request merged**